src/Server/RestRouterHandlers/TableRestRouterHandler.cpp

```diff
@@ -326,4 +345,11 @@ String TableRestRouterHandler::buildResponse() const
     return resp_str_stream.str();
 }
 
+bool TableRestRouterHandler::tableExists(const String & database, const String & table) const
+{
+    const auto & catalog_service = CatalogService::instance(query_context);
```

Comment:
Let's implement `tableExists` in CatalogService, in that way we don't need return the table definition

Reply:
fixed